File: VarsityCode2023/Challenge 4/solution2.js
'use strict';

module.exports = {
  balancePoint
};

function balancePoint(input) {
    // Your code goes here
    let leftSum = 0;
    let rightSum = input.reduce((a,b) => a+b, 0);
    
    for (let i = 0; i < input.length; i++) {
        rightSum -= input[i];
        if (leftSum === rightSum){
            return i;
        }
        leftSum += input[i];
    }
    return -1
}

File: VarsityCode2023/Challenge 4/solution2.py
def find_balance_point(lst):
    left_sum = 0
    right_sum = sum(lst)

    for i in range(len(lst)):
        right_sum -= lst[i]

        if left_sum == right_sum:
            return i

        left_sum += lst[i]

    return -1
